pkg/reconciler/openshift/tektonhub/extension.go

```diff
@@ -116,6 +128,34 @@ func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.Tekto
 		return err
 	}
 
+	// Create UI route based on the value of ui i.e. false/true
+
+	uiHubDir := filepath.Join(common.ComponentDir(th), common.TargetVersion(th), tektonHubUiResourceKey)
+	uiManifest := oe.manifest.Append()
+
+	if err := common.AppendManifest(&uiManifest, uiHubDir); err != nil {
+		return err
+	}
+	uiManifest, err = uiManifest.Transform(
+		mf.InjectOwner(th),
+		mf.InjectNamespace(targetNs),
+	)
+	if err != nil {
+		logger.Error("failed to transform manifest")
+		return err
+	}
+
+	if err := uiManifest.Filter(mf.ByKind("Route")).Apply(); err != nil {
+		return err
+	}
```

Comment:
```suggestion
        routeManifest := uiManifest.Filter(mf.ByKind("Route"))
	routeManifest, err = routeManifest.Transform(
		mf.InjectOwner(th),
		mf.InjectNamespace(targetNs),
	)
	if err != nil {
		logger.Error("failed to transform manifest")
		return err
	}

	if err := routeManifest.Apply(); err != nil {
		return err
	}
```

how about this, as we are only creating the routes in here

Reply:
same applies to lines 95 for API route